phantom-state benchmark build failure

Ping @Daniel-Diaz

```
Building phantom-state-0.2.1.1...
Preprocessing library phantom-state-0.2.1.1...
[1 of 1] Compiling Control.Applicative.PhantomState ( Control/Applicative/PhantomState.hs, dist/build/Control/Applicative/PhantomState.o )
Preprocessing benchmark 'phantom-state-bench' for phantom-state-0.2.1.1...
[1 of 1] Compiling Main             ( bench/Main.hs, dist/build/phantom-state-bench/phantom-state-bench-tmp/Main.o )

bench/Main.hs:35:49: error:
    • Couldn't match type ‘s’
                     with ‘primitive-0.6.1.0:Control.Monad.Primitive.PrimState m0’
      ‘s’ is a rigid type variable bound by
        a type expected by the context:
          forall s. GHC.ST.ST s (MV.MVector s Int)
        at bench/Main.hs:33:19
      Expected type: MV.MVector
                       (primitive-0.6.1.0:Control.Monad.Primitive.PrimState m0) Int
        Actual type: MV.MVector s Int
    • In the first argument of ‘MV.write’, namely ‘v’
      In the first argument of ‘(>>)’, namely ‘MV.write v i i’
      In the expression: MV.write v i i >> pure (i + 1)
    • Relevant bindings include
        loop :: PhantomStateT Int m0 () (bound at bench/Main.hs:35:7)
        v :: MV.MVector s Int (bound at bench/Main.hs:34:3)

bench/Main.hs:36:35: error:
    Variable not in scope:
      step :: PhantomStateT Integer (GHC.ST.ST s) a0
```

--- a/build-constraints.yaml
+++ b/build-constraints.yaml
@@ -2857,9 +2857,6 @@ expected-benchmark-failures:
 
     # GHC 8
     - cassava
-
-    # https://github.com/fpco/stackage/issues/1539
-    - phantom-state
 # end of expected-benchmark-failures
 
 

